src/utils/modexp/modexp_utils.cairo

```diff
@@ -0,0 +1,66 @@
+from starkware.cairo.common.cairo_builtins import BitwiseBuiltin
+from starkware.cairo.common.uint256 import (
+    Uint256,
+    uint256_add,
+    uint256_and,
+    uint256_check,
+    uint256_mul_div_mod,
+    uint256_mul,
+    uint256_eq,
+    uint256_lt,
+    uint256_shr,
+    uint256_sub,
+    uint256_unsigned_div_rem,
+)
+from starkware.cairo.common.bitwise import bitwise_and
+from starkware.cairo.common.registers import get_label_location
+
+// @title ModExpHelper Functions
+// @notice This file contains a selection of helper functions for modular exponentiation and gas cost calculation.
+// @author @dragan2234
+// @custom:namespace ModExpHelpers
+namespace ModExpHelpers {
+    /// @title Modular exponentiation calculation
+    /// @author dragan2234
+    /// @dev Computes x ** y % p for Uint256 numbers via fast modular exponentiation algorithm.
+    /// Time complexity is log_2(y).
+    /// Loop is implemented via uint256_modexp_recursive_call() function.
+    func uint256_modexp{range_check_ptr: felt}(x: Uint256, y: Uint256, p: Uint256) -> (
```

Comment:
according to you naming convention (namespace is `ModExpHelpers`) and starkware oen (e.g. `uint256_mul_div_mod`) I guess that this should be `uint256_mod_exp`